### PR TITLE
Add Spellchecking to CI

### PR DESCRIPTION
**Pull request by Naraenda**
**Scope**
- [x] Adds spellchecking to CI.
- [x] Word list for the `rocm-docs-core` and `ROCm` repositories.
- [x] Fixed spelling mistakes.

Related issue: https://github.com/RadeonOpenCompute/ROCm/issues/1988

### CODE REVIEW

**Comment by saadrahim** [comment]
CI failure needs resolution

**Comment by Naraenda** [comment]
> CI failure needs resolution

Fixed in: https://github.com/RadeonOpenCompute/rocm-docs-core/pull/116

**Comment by Naraenda** [comment]
Some other (spelling/style) considerations:
- "frontend" vs "front-end" vs "front end".
  - My spellchecker doesn't like the first. The middle one is adjective. The last one is more common is formal and technical writing.
  - Similar for "back end".
- "pretrained" vs "pre-trained". Context: AI.
  - The latter is used more in a formal settings.
- However between "preprocess" (without the hyphen) is more used.
  - Do want something more strict and keep consistent, or allow more freedom?
- All file paths (`path/to/file`), executable programs (`gcc`), and libraries/packages (`rocm-docs-core`) should be embedded as \`text\`.
  - These get automatically ignored by the spelling  checker.
  - However projects that have proper names should be written like: ROCm, PyTorch, AMD, LLVM. These should be added to the word list if needed.
- Should it be "AMDGCN" or "AMD GCN"?
- What about "AMDGPU"?

Honestly most of this is just nitpicking, but I think the last few are important that should be added to the documentation guidelines.

File: .wordlist.txt [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+CPUs
+CSE
+CSV
+CU
+cuBLAS
+CUDA
+cuLIB
+cuRAND
+CUs
+dataset
```
Comment (saadrahim) [inline .wordlist.txt:46]: duplicate

**Pull request merged**